test/integration/integration_test.go

```diff
@@ -200,3 +173,68 @@ func testMultipleIngresses(t *testing.T) {
 		}
 	}
 }
+
+type TestConfig struct {
+	operatorNamespace string
+	clusterName       string
+
+	kubeConfig string
+
+	t *testing.T
+}
+
+func NewTestConfig(t *testing.T) *TestConfig {
+	config := &TestConfig{t: t}
+
+	// Check prerequisites
+	kubeConfig := os.Getenv("KUBECONFIG")
+	if len(kubeConfig) == 0 {
+		t.Fatalf("KUBECONFIG is required")
+	}
+	// The operator-sdk uses KUBERNETES_CONFIG...
+	os.Setenv("KUBERNETES_CONFIG", kubeConfig)
+	config.kubeConfig = kubeConfig
+
+	clusterName := os.Getenv("CLUSTER_NAME")
+	if len(clusterName) == 0 {
+		t.Fatalf("CLUSTER_NAME is required")
+	}
+	config.clusterName = clusterName
+
+	namespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		namespace = "default"
+		os.Setenv("WATCH_NAMESPACE", namespace)
+	}
+	config.operatorNamespace = namespace
+
+	return config
+}
+
+func (c *TestConfig) startOperator() {
+	resource := "ingress.openshift.io/v1alpha1"
+	kind := "ClusterIngress"
+	resyncPeriod := 5
+	c.t.Logf("Watching %s, %s, %s, %d", resource, kind, c.operatorNamespace, resyncPeriod)
+	sdk.Watch(resource, kind, c.operatorNamespace, resyncPeriod)
+	sdk.Handle(stub.NewHandler())
+	go sdk.Run(context.TODO())
+}
+
+func (tc *TestConfig) createCRD() {
+	tc.runShellCmd(fmt.Sprintf("oc apply -f ../../deploy/crd.yaml -n %s", tc.operatorNamespace), "create cluster ingress CRD")
+}
+
+func (tc *TestConfig) deleteCRD() {
+	tc.runShellCmd(fmt.Sprintf("oc delete crd clusteringresses.ingress.openshift.io -n %s", tc.operatorNamespace), "delete cluster ingress CRD")
+}
+
+func (c *TestConfig) runShellCmd(command, msg string) {
```

Comment:
This is fine for now, but I think we should consider wrapping this in a script that invokes our tests, or compiling the manifests into bindata and using our Go client to apply them.

Reply:
Any yaml that doesn't need runtime change, we can run directly using cli command and for others we can generate objects.